Source/DafnyLanguageServer/Workspace/DocumentDatabase.cs

```diff
@@ -213,20 +214,38 @@ IRelocator relocator
 
     public async Task<DafnyDocument?> GetVerifiedDocumentAsync(TextDocumentIdentifier documentId) {
       if (documents.TryGetValue(documentId.Uri, out var databaseEntry)) {
-        return await databaseEntry.VerifiedDocument;
+        return await databaseEntry.FullyVerifiedDocument;
       }
       return null;
     }
 
     public IReadOnlyDictionary<DocumentUri, IDocumentEntry> Documents =>
       documents.ToDictionary(k => k.Key, v => (IDocumentEntry)v.Value);
 
-    private record DocumentEntry(int? Version, Task<DafnyDocument> ResolvedDocument,
-      Task<DafnyDocument> VerifiedDocument,
-      CancellationTokenSource CancellationSource) : IDocumentEntry {
+    private class DocumentEntry : IDocumentEntry {
+      private readonly CancellationTokenSource cancellationSource;
+      public int? Version { get; }
+      public Task<DafnyDocument> ResolvedDocument { get; }
+
+      public DocumentEntry(int? version, Task<DafnyDocument> resolvedDocument,
+        IObservable<DafnyDocument> verifiedDocuments,
+        CancellationTokenSource cancellationSource) {
+        this.cancellationSource = cancellationSource;
+        Version = version;
+        ResolvedDocument = resolvedDocument;
+        MostVerifiedDocument = resolvedDocument;
```

Comment:
I see that you are trying to align all the concepts of the pipeline in a most up-to-date document, and that "verification" is the thing that goes from 0 to 100%. I'm not sure about that but I think that's fine for now. I would have put `MostUptoDateDocument` but that's just a naming convention and yours can be defended, so let's keep it that way.